File: Two-Pointer/numRescueBoats.py
from typing import List

class Solution:
    def numRescueBoats(self, people: List[int], limit: int) -> int:
        people.sort()
        low = 0
        high = len(people) - 1
        result = 0

        while low < high:
            if people[low] + people[high] <= limit:
                result += 1
                low += 1
                high -= 1
            else:
                result += 1
                high -= 1
        
        if low == high: result += 1

        return result

File: Two-Pointer/numRescueBoats.js
function numRescueBoats(people, limit) {
  let result = 0;
  let i = 0;
  let j = people.length - 1;

  people.sort((a, b) => a - b);

  while (i <= j) {
    if (people[i] + people[j] <= limit) {
      result++;
      i++;
      j--;
    } else {
      result++;
      j--;
    }
  }

  return result;
}
